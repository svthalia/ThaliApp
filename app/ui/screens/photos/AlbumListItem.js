import React from 'react';
import PropTypes from 'prop-types';
import {
  ImageBackground, Text, TouchableHighlight, View, ViewPropTypes,
} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import Moment from 'moment/moment';
import PhotoListItem from './PhotoListItem';
import styles from './style/AlbumListItem';
import SquareView from '../../components/memberView/SquareView';

const AlbumListItem = (props) => {
  const dateText = Moment(props.album.date).format('DD-MM-YY');

  return (
    <SquareView style={props.style} size={props.size}>
      <TouchableHighlight
        style={styles.image}
        onPress={() => props.openAlbum(props.album.pk)}
      >
        <ImageBackground
          style={styles.image}
          source={{
            uri: props.album.cover.file.small,
            headers: {
              Authorization: `Token ${props.token}`,
            },
          }}
        >
          <LinearGradient colors={['#55000000', '#000000']} style={styles.overlayGradient} />
          <View style={styles.textWrapper}>
            <Text style={styles.titleText}>
              {props.album.title}
            </Text>
            <Text style={styles.dateText}>
              {dateText}
            </Text>
          </View>
        </ImageBackground>
      </TouchableHighlight>
    </SquareView>
  );
};

AlbumListItem.propTypes = {
  size: PropTypes.number.isRequired,
  style: ViewPropTypes.style,
  album: PropTypes.shape({
    pk: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    shareable: PropTypes.bool.isRequired,
    accessible: PropTypes.bool.isRequired,
    hidden: PropTypes.bool.isRequired,
    cover: PropTypes.shape(PhotoListItem.propTypes.photo),
  }).isRequired,
  openAlbum: PropTypes.func.isRequired,
  token: PropTypes.string.isRequired,
};

AlbumListItem.defaultProps = {
  style: {},
};

export default AlbumListItem;
