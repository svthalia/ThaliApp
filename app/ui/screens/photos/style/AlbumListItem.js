import { StyleSheet } from 'react-native';
import Colors from '../../../style/Colors';

const styles = StyleSheet.create({
  textWrapper: {
    position: 'absolute',
    left: 6,
    bottom: 4,
    right: 6,
  },
  titleText: {
    fontSize: 12,
    color: Colors.white,
    backgroundColor: Colors.transparent,
  },
  dateText: {
    fontSize: 10,
    color: Colors.white,
    backgroundColor: Colors.transparent,
  },
  image: {
    flex: 1,
  },
  overlayGradient: {
    position: 'absolute',
    top: '50%',
    bottom: 0,
    left: 0,
    right: 0,
    opacity: 0.6,
  },
});

export default styles;
