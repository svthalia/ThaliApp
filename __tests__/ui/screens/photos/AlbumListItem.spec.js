import React from 'react';
import renderer from 'react-test-renderer';
import AlbumListItem from '../../../../app/ui/screens/photos/AlbumListItem';

describe('AlbumListItem component', () => {
  const photo = {
    file: {
      small: 'http://test.local/small.png',
      medium: 'http://test.local/medium.png',
      large: 'http://test.local/large.png',
      full: 'http://test.local/full.png',
    },
    album: 0,
    pk: 0,
    size: 10,
  };
  const album = {
    pk: 1,
    title: 'Title',
    date: '01-01-2017',
    cover: photo,
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<AlbumListItem openAlbum={() => {}} album={album} size={20} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
