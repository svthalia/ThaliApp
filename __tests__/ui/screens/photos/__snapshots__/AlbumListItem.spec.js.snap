// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumListItem component renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "height": 20,
        "width": 20,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "http://test.local/small.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <BVLinearGradient
        colors={
          Array [
            5570560,
            4278190080,
          ]
        }
        endPoint={undefined}
        locations={null}
        startPoint={undefined}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0.6,
            "position": "absolute",
            "right": 0,
            "top": "50%",
          }
        }
      />
      <View
        style={
          Object {
            "bottom": 4,
            "left": 6,
            "position": "absolute",
            "right": 6,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontSize": 12,
            }
          }
        >
          Title
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#FFFFFF",
              "fontSize": 10,
            }
          }
        >
          01-01-17
        </Text>
      </View>
    </View>
  </View>
</View>
`;
